### ISSUE TITLE
AD Import 

### ISSUE THREAD

**Issue by sandeep0308** [Kentico/ADImport#8]
Hello
I am trying to install Active directory and import all the users and roles into Kentico, Could anyone walk me through the steps you followed to accomplish it? I installed ADImport.exe and stuck at the errors I have been receiving.  I don't see ADImport tool in Kentico website though, since it's a new tool there is no clear explanation I could find in any of the sites.

**Comment by petrsvihlik** [comment]
Hi @sandeep0308 , have you read the documentation? https://github.com/Kentico/ADImport/wiki
What kind of errors are you getting? Can you share the messages / screenshots with us?

**Comment by sandeep0308** [comment]
Hi @petrsvihlik 
Thanks for prompt response. After running ADImport.exe from the downloaded folder, I try establishing Database connection and clicked next, it's throwing these two errors. I tried this using both trial and licensed version.
System.IO.FileNotFoundException: Could not load file or assembly 'Mono.Cecil, Version=0.9.6.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756' or one of its dependencies. The system cannot find the file specified.
File name: 'Mono.Cecil, Version=0.9.6.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756'
   at CMS.Core.AssemblyDiscovery.IsAssemblyDiscoverable(String filePath)
   at CMS.Core.AssemblyDiscovery.AddOnlyDiscoverableAssembly(ICollection`1 assemblies, Dictionary`2 preloadedAssemblies, AssemblyName assemblyName, String filePath)
   at CMS.Core.AssemblyDiscovery.ExecuteGetAssemblies(Boolean onlyDiscoverable)
   at CMS.Core.AssemblyDiscovery.get_DiscoverableAssemblies()
   at CMS.Core.TypeManager.PreInitializeTypes()
   at CMS.Core.ModuleEntryManager.PreInit()
   at CMS.Core.AppCore.PreInit()
   at CMS.DataEngine.CMSApplication.PreInit()
   at CMS.DataEngine.CMSApplication.Init()
   at ADImport.Step2.ValidateDatabase() in C:\ ADImport\Steps\Step2.cs:line 379
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ADImport.Step2.<cmbDBName_SelectedIndexChanged>d__15.MoveNext() in C:\ ADImport\Steps\Step2.cs:line 156
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].



************** Loaded Assemblies **************
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.2046.0 built by: NET47REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.2102.0 built by: NET47REL1LAST
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.2046.0 built by: NET47REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
CMS.Base
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.6174.20089
    CodeBase: file:///C:/Program%20Files/Kentico/10.0/Bin/ADImport-v10.0.0/CMS.Base.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

****************************************************************************************************************************************************





See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.InvalidOperationException: A loop occurred in application core pre-initialization. This is likely a result of improper pre-initialization in some module.
   at CMS.Core.AppCore.PreInit()
   at CMS.DataEngine.CMSApplication.PreInit()
   at CMS.DataEngine.CMSApplication.Init()
   at ADImport.Step2.ValidateDatabase() in C:\ADImport\Steps\Step2.cs:line 379
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ADImport.Step2.<cmbDBName_SelectedIndexChanged>d__15.MoveNext() in C:\ ADImport\Steps\Step2.cs:line 156
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.2102.0 built by: NET47REL1LAST
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
ADImport
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.0.0
    CodeBase: file:///C://Downloads/ADImport-v10.0.0/ADImport.exe

**Comment by petrsvihlik** [comment]
Could you verify that you have extracted all the files including the CMSDependencies folder?


![Imgur](https://i.imgur.com/ac6SVwC.png)

**Comment by ZdenekS81** [comment]
Hi @sandeep0308, is the issue still valid or did the last comment help you to fix it, please?

**Comment by hvanpelt1129** [comment]
I'm having the same issue after upgrading from Kentico 9 to 10.  I do have CMSDependencies folder and have verified the the Mono.Cecil.0.9.6.0 folder and Mono.Cecil.dll file is there.

**Comment by twbocklertorch** [comment]
I'm running into the exact same error as @sandeep0308 .  I tried on the web server (Windows Server 2012R2) and also on my local machine (Windows 10).  I ran the .exe as administrator, as well.  Neither way worked.  I receive this error when I selected the database (after establishing a SQL Server connection).  I am running the v10.0 version as that is the version of Kentico installed.
![kentico_ad_error](https://user-images.githubusercontent.com/21347874/35397692-bbf4ffa4-01b5-11e8-8d02-47970d816092.png)
